### PR TITLE
feat: ITO qualification

### LINKED ISSUE #3065: [Demand] Show Qualification Result
## Catalogue

- [ ] UI/UX
- [ ] Mask Core Feature
- [x] Mask Plugin Feature (ITO)
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

It should show the qualification result message on the submit button if the currently selected account is not allowed to attend the ITO.

## Part 2: Problem to Solve

Write the qualification failed message on the `Enter` button.

image

## Notes

+ A deny all address contract on Ropsten Network: https://ropsten.etherscan.io/address/0x0259644b7f5f590be16e849637d28c0bc4511bb0#readContract
+ A deny all ITO on Twitter: https://twitter.com/differui/status/1388010303938449410

### PR DESCRIPTION
**Pull request by guanbinrui**
closes #3065

image
image
image

### CODE REVIEW

File: packages/maskbook/src/plugins/ITO/UI/ITO.tsx [hunk 1]
```diff
@@ -612,6 +622,10 @@ export function ITO(props: ITO_Props) {
                         className={classes.actionButton}>
                         {t('plugin_ito_withdraw')}
                     </ActionButton>
+                ) : !ifQualified ? (
```
Comment (zhouhanseng) [inline packages/maskbook/src/plugins/ITO/UI/ITO.tsx:625]: Put this button side by side with enter swap button. Assume someone swapped MASK, then spend them all, he/she should still be able to claim.

Comment (zhouhanseng) [reply packages/maskbook/src/plugins/ITO/UI/ITO.tsx:625]:
```suggestion
) : !ifQualified && !isBuyer ? (
```

Comment (guanbinrui) [reply packages/maskbook/src/plugins/ITO/UI/ITO.tsx]: Yes, it should not verify the qualification of claimers.

**Review by zhouhanseng — APPROVED** [review]

File: packages/maskbook/src/plugins/ITO/UI/ITO.tsx [hunk 2]
```diff
@@ -763,10 +782,7 @@ function ITO_LoadingFailUI({
 
 export function ITO_ConnectMetaMask() {
     return (
-        <ITO_LoadingFailUI
-            retryPoolPayload={async () => await Services.Ethereum.connectMetaMask()}
-            isConnectMetaMask={true}
-        />
+        <ITO_LoadingFailUI retryPoolPayload={async () => await Services.Ethereum.connectMetaMask()} isConnectMetaMask />
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/ITO.tsx:785]: extract `retryPoolPayload` property as variable
```suggestion
        const onRetry = () => Services.Ethereum.connectMetaMask()
        return <ITO_LoadingFailUI retryPoolPayload={onRetry} isConnectMetaMask />
```

Comment (guanbinrui) [reply packages/maskbook/src/plugins/ITO/UI/ITO.tsx:785]: Maybe another PR.

**Review by UncleBill — APPROVED** [review]

File: packages/maskbook/src/plugins/ITO/UI/ITO.tsx (hunk 2 above)
Comment (septs) [reply packages/maskbook/src/plugins/ITO/UI/ITO.tsx:785]:
```suggestion
        <ITO_LoadingFailUI retryPoolPayload={() => Services.Ethereum.connectMetaMask()} isConnectMetaMask />
```

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**